kp_test.go

```diff
@@ -729,6 +759,31 @@ func TestKeys(t *testing.T) {
 				return nil
 			},
 		},
+		{
+			"Restore Key",
+			func(t *testing.T, api *API, ctx context.Context) error {
+				MockAuthURL(keyURL, http.StatusCreated, testKeys)
+				k, err := api.CreateImportedRootKey(ctx, "test", nil, "asdfqwerasdfqwerasdfqwerasdfqwer", "", "")
+				assert.NoError(t, err)
+
+				key1 := k.ID
+
+				MockAuthURL(keyURL, http.StatusOK, testDeletedKey)
+				kd, err := api.DeleteKey(ctx, key1, ReturnRepresentation)
+				assert.NoError(t, err)
+				assert.Equal(t, kd.State, 5)
+
+				MockAuthURL(keyURL, http.StatusOK, testRestoredKey)
+				kr, err := api.RestoreKey(ctx, key1, "JaokkJZffuuMOOC4YhuFspe8508ixeKvqskKhFw1f+w=", "", "", ReturnRepresentation)
+				assert.NoError(t, err)
+				assert.Equal(t, kr.State, 1)
+
+				MockAuthURL(keyURL, http.StatusServiceUnavailable, "{}")
+				_, err = api.RestoreKey(ctx, testKey, "JaokkJZffuuMOOC4YhuFspe8508ixeKvqskKhFw1f+w=", "", "", ReturnMinimal)
```

Comment:
You are not testing anything except the stubbed out call here... please remove this test and write one that follows the style of the ones below.

You need to check what gets sent in the request and response instead of just checking if the error is nil.